koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
             'repositories': all_repositories,
             'koji_builds': all_koji_builds,
         }
+
+
+class BuildSourceContainerTask(BaseContainerTask):
+    # Start builds via osbs
+    Methods = ['buildSourceContainer']
+    # Same value as for regular 'build' method.
+    _taskWeight = 2.0
+
+    # JSON Schema definition for koji buildSourceContainer task parameters
+    # Used to validate arguments passed to the handler() method of this class
+    PARAMS_SCHEMA = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "description": "Parameters for a koji buildSourceContainer task.",
+
+        "type": "array",
+        "items": [
+            {
+                "type": "string",
+                "description": "Build target."
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources"
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources"
+                    },
+                    "signing_intent": {
+                        "type": ["string"],
+                        "description": "Signing intent of the ODCS composes."
+                    }
+                },
+                "oneOf": [{
+                    "required": ["koji_build_nvr"]
+                }, {
+                    "required": ["koji_build_id"]
+                }],
+                "additionalProperties": "false"
+            }
+        ],
+        'minItems': 2
+    }
+
+    def __init__(self, id, method, params, session, options, workdir=None, demux=False):
+        BaseContainerTask.__init__(self, id, method, params, session, options,
+                                   workdir)
+        self._osbs = None
+        self.demux = demux
```

Comment:
Maybe we want to always leave this as `False`?

Reply:
well, we have it default to False in init and we aren't passing it anywhere